### PR TITLE
Add detail

### PR DESCRIPTION
**Pull request by erikfloresq**
- Add React Router
- Add detail/:countryName path

https://github.com/user-attachments/assets/b167c971-5249-48f1-8cb3-78a0c59f2701

### CODE REVIEW

File: src/app/App.tsx [hunk 1]
```diff
@@ -36,8 +37,11 @@ function App() {
             .filter((country) =>
               country.name.toLowerCase().includes(searchText)
             )
-            .map((country) =>  
-              <Card country={country} />
+            .map((country) =>
+              <Link to={ `detail/${country.name.toLowerCase()}` }>
```
Comment (VictorJSV) [inline src/app/App.tsx]: we forget the key prop:
https://react.dev/learn/rendering-lists#keeping-list-items-in-order-with-key

File: src/pages/Detail.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+export const loader = async ({ params }: any) => {
+    const country = await getCountry(params.countryName);
+    return country[0];
+}
+
+const Detail = () => {
+    const country = useRouteLoaderData('detail') as Country;
+
+    let borders = country.borders?.map((border) => (
+        <li>
```
Comment (VictorJSV) [inline src/pages/Detail.tsx]: key prop

File: src/pages/Detail.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import React from "react";
+import { useRouteLoaderData } from "react-router-dom";
+import { getCountry } from '../app/countries';
+import type { Country } from '../models/Country';
+import "./Detail.css";
+
+export const loader = async ({ params }: any) => {
```
Comment (VictorJSV) [inline src/pages/Detail.tsx:7]: any xD

File: src/pages/Detail.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                        {country.flag}
+                    </li>
+                    <li>
+                        <label>Timezones:</label>
+                        {country.timezones}
+                    </li>
+                    <li>
+                        <label>Borders:</label>
+                        <ul>
+                            {borders}
```
Comment (VictorJSV) [inline src/pages/Detail.tsx]: Also, could be this:
```

{country.borders?.map(border) => (

  <> ... </>

)}
```

File: src/pages/Detail.tsx (hunk 3 above)
Comment (erikfloresq) [reply src/pages/Detail.tsx:7]: I didn't found the type sor this 😭

**Pull request merged**